### PR TITLE
Allow certificate data access in plugins

### PR DESCRIPTION
**Pull request by matiasinsaurralde**
For #1772, still need to refactor certificate meta to see how it can be added to the request object.
This allows raw certificate data access.

### CODE REVIEW

**Comment by buger** [comment]
You have type "bytes" for this record, but in fact this field can contain multiple certificates. Should not it be array of bytes instead? 

As for Certificate Meta, maybe just serialize to JSON? Also to avoid overhead, I suggest adding new manifest variable `require_client_certificate`, similar to `require_session`, and only if its true, it will load this variables.

File: coprocess_grpc_test.go [hunk 1]
```diff
@@ -204,26 +208,28 @@ func loadTestGRPCAPIs() {
 	})
 }
 
-func startTykWithGRPC() (*tykTestServer, *grpc.Server) {
+func startTykWithGRPC(grpcPort int) (*tykTestServer, *grpc.Server) {
 	// Setup the gRPC server:
-	listener, _ := net.Listen("tcp", grpcListenAddr)
+	listener, _ := net.Listen("tcp", fmt.Sprintf(":%d", grpcPort))
```
Comment (lsjurczak) [inline coprocess_grpc_test.go]: It's better to use `strconv.Itoa`. Sprintf will use the package reflect, and it will allocate one more object.

**Comment by buger** [comment]
@matiasinsaurralde bump

**Comment by matiasinsaurralde** [comment]
Updated with above changes and tests!

**Comment by buger** [comment]
Nice! Do you think we can make it available to JSVM plugins too?

**Comment by marksou** [comment]
I'm assuming the docs will need updating?

**Comment by buger** [comment]
yes, I added "needs documenting" label for the task. 
It will be released as part of 2.8.1 and 2.7.5

**Comment by matiasinsaurralde** [comment]
I will rebase this (we updated the PB bindings recently) and add JSVM support, should be possible!

**Pull request closed without merging**